api/src/backend/api/v1/serializer_utils/base.py

```diff
@@ -0,0 +1,23 @@
+import yaml
+from rest_framework_json_api import serializers
+from rest_framework_json_api.serializers import ValidationError
+
+
+class BaseValidateSerializer(serializers.Serializer):
+    def validate(self, data):
+        if hasattr(self, "initial_data"):
+            initial_data = set(self.initial_data.keys()) - {"id", "type"}
+            unknown_keys = initial_data - set(self.fields.keys())
+            if unknown_keys:
+                raise ValidationError(f"Invalid fields: {unknown_keys}")
+        return data
+
+
+class YamlOrJsonField(serializers.JSONField):
```

Comment:
I'm a little bit confused about what this does. I need to continue the review but JSON and YAML are interoperable, so in the SDK the Mutelist is input as YAML but then transformed to JSON to work with it.

Not sure if this comment applies, but just in case.